client/components/data/query-site-purchases/index.jsx

```diff
@@ -12,7 +12,9 @@ import { fetchSitePurchases } from 'state/purchases/actions';
 
 class QuerySitePurchases extends Component {
 	requestSitePurchases( props = this.props ) {
-		this.props.fetchSitePurchases( props.siteId );
+		if ( props.siteId ) {
```

Comment:
If we're making this one optional, it would probably be a good idea to change this prop type in `propTypes` to be non-required.